src/main/java/minicraft/item/TileItem.java

```diff
@@ -23,90 +23,109 @@ protected static ArrayList<Item> getAllInstances() {
-		items.add(new TileItem("hole", SpriteLinker.missingTexture(SpriteType.Item), "hole", "dirt", "grass"));
-		items.add(new TileItem("lava", SpriteLinker.missingTexture(SpriteType.Item), "lava", "dirt", "grass", "hole"));
-		items.add(new TileItem("path", SpriteLinker.missingTexture(SpriteType.Item), "path", "dirt", "grass", "hole"));
-		items.add(new TileItem("water", SpriteLinker.missingTexture(SpriteType.Item), "water", "dirt", "grass", "hole"));
+		items.add(new TileItem("Farmland", SpriteLinker.missingTexture(SpriteType.Item), new TileModel("farmland"), "dirt", "grass", "hole"));
+		items.add(new TileItem("hole", SpriteLinker.missingTexture(SpriteType.Item), new TileModel("hole"), "dirt", "grass"));
+		items.add(new TileItem("lava", SpriteLinker.missingTexture(SpriteType.Item), new TileModel("lava"), "dirt", "grass", "hole"));
+		items.add(new TileItem("path", SpriteLinker.missingTexture(SpriteType.Item), new TileModel("path"), "dirt", "grass", "hole"));
+		items.add(new TileItem("water", SpriteLinker.missingTexture(SpriteType.Item), new TileModel("water"), "dirt", "grass", "hole"));
 
 		return items;
 	}
 
-	public final String model;
+	public final TileModel model;
 	public final List<String> validTiles;
 
-	protected TileItem(String name, LinkedSprite sprite, String model, String... validTiles) {
+	protected TileItem(String name, LinkedSprite sprite, TileModel model, String... validTiles) {
 		this(name, sprite, 1, model, Arrays.asList(validTiles));
 	}
-	protected TileItem(String name, LinkedSprite sprite, int count, String model, String... validTiles) {
+	protected TileItem(String name, LinkedSprite sprite, int count, TileModel model, String... validTiles) {
 		this(name, sprite, count, model, Arrays.asList(validTiles));
 	}
-	protected TileItem(String name, LinkedSprite sprite, int count, String model, List<String> validTiles) {
+	protected TileItem(String name, LinkedSprite sprite, int count, TileModel model, List<String> validTiles) {
 		super(name, sprite, count);
-		this.model = model.toUpperCase();
+		this.model = model;
 		this.validTiles = new ArrayList<>();
 		for (String tile: validTiles)
 			 this.validTiles.add(tile.toUpperCase());
 	}
 
+	public static class TileModel {
+		public static final TileDataGetter DEFAULT_DATA = ((model, target, level, xt, yt, player, attackDir) -> model.getDefaultData());
+		public static final TileDataGetter KEEP_DATA = ((model, target, level, xt, yt, player, attackDir) -> level.getData(xt, yt));
+
+		public final String tile;
+		public final TileDataGetter data;
+
+		@FunctionalInterface
+		interface TileDataGetter {
+			int getTileData(Tile model, Tile target, Level level, int xt, int yt, Player player, Direction attackDir);
+		}
+
+		public TileModel(String tile) { this(tile, DEFAULT_DATA); }
+		public TileModel(String tile, TileDataGetter data) {
+			this.tile = tile.toUpperCase();
+			this.data = data;
+		}
+	}
+
```

Comment:
This seems rather pointless.

Reply:
`TileModel` is kept for more functional use as the tile data might be impacted by the target tile.

Reply:
I do like the game using less than 1 GB of ram.

Reply:
Would they really use that amount of memory?

Reply:
Yes, we're talking about several bytes of data for over a thousand tiles.

Reply:
That class is used for items only.

Reply:
Also, this class can prevent making new subclasses of `TileItem`.

Reply:
For each item then. That's still a lot of data.

Reply:
The instances are still small comparing with plane `String`.

Reply:
```java
	private static final ArrayList<Item> items = new ArrayList<>();

	public static void main(String[] args) {
		Item item = getAllInstances().get(0);
		for (int i = 0; i < 1000; i++)
			items.add(item.clone());
		while (true) {}
	}
```
`TileItem` using `TileModel`:

![Eclipse Memory Analyzer 2023_02_26 12_08_12](https://user-images.githubusercontent.com/74168521/221391948-c324f970-ce11-4600-9e4a-e20a94457d2a.png)

`TileItem` using direct(/plane) `String`

![Eclipse Memory Analyzer 2023_02_26 12_08_16](https://user-images.githubusercontent.com/74168521/221391979-c82abc23-049d-427a-bff2-9b951eac4232.png)

I do not think that that is `a lot of data`.